ui/app/AppLayouts/Shell/ShellAdaptor.qml

```diff
@@ -0,0 +1,401 @@
+        }
+        expectedRoles: ["url", "name", "iconUrl", "connectorBadge"]
+        exposedRoles: ["key", "id", "name", "icon", "color", "connectorBadge"]
+    }
+
+    ConcatModel {
+        id: combinedModel
+        sources: [
+            SourceModel {
+                model: root.showCommunities ? communitiesModel : null
+                markerRoleValue: Constants.appSection.community
+            },
+            SourceModel {
+                model: root.showWallets ? walletsModel : null
+                markerRoleValue: Constants.appSection.wallet
+            },
+            SourceModel {
+                model: root.showSettings ? settingsModel : null
+                markerRoleValue: Constants.appSection.profile
+            },
+            SourceModel {
+                model: root.showChats ? chatsModel : null
+                markerRoleValue: Constants.appSection.chat
+            },
+            SourceModel {
+                model: root.showDapps ? dappsModel : null
+                markerRoleValue: Constants.appSection.dApp
+            }
+        ]
+
+        markerRoleName: "sectionType"
+        expectedRoles: ["key", "id", "enabled", "name", "icon", "color", "hasNotification", "notificationsCount", // common props
+            "banner", "members", "activeMembers", "pending", "banned", // community
+            "isExperimental", // settings
+            "walletType", "currencyBalance", // wallet
+            "connectorBadge" // dapp
+        ]
+    }
+
+    Settings {
+        id: shellSettings
+        category: "Shell_%1".arg(root.profileId)
+    }
+
+    ObjectProxyModel { // provides a writable overlay for "timestamp" and "pinned" roles
+        id: shellProxyModel
+
+        sourceModel: combinedModel
+        delegate: QtObject {
+            property real timestamp
+            property bool pinned
+        }
+
+        exposedRoles: ["timestamp", "pinned"]
+    }
+
+    function setPinned(key, pinned) {
+        const idx = ModelUtils.indexOf(shellProxyModel, "key", key)
+        if (idx > -1) {
+            shellProxyModel.proxyObject(idx).pinned = pinned
+        }
+    }
+
+    function setTimestamp(key, timestamp) {
+        const idx = ModelUtils.indexOf(shellProxyModel, "key", key)
+        if (idx > -1) {
+            shellProxyModel.proxyObject(idx).timestamp = timestamp
+        }
+    }
+
+    function save() {
+        const dataArray = ModelUtils.modelToArray(shellProxyModel, ["key", "timestamp", "pinned"])
+        const settingsData = JSON.stringify(dataArray)
+        shellSettings.setValue("ShellEntries", settingsData)
+        shellSettings.sync()
+    }
+
+    function load() {
+        var length = 0
+        const settingsData = shellSettings.value("ShellEntries")
+        var dataArray = []
+
+        try {
+            dataArray = JSON.parse(settingsData)
+            length = dataArray.length
+        } catch (e) {
+            return
+        }
+
+        dataArray.forEach(function(item) {
```

Comment:
nitpick:
```suggestion
        dataArray.forEach(item =>  {
```

Reply:
Yeah... but it just messes up the indentation, in fact that's the only reason I use the longer `function()` syntax